### DISCUSSION TITLE
Vercel dev command uses a proxy

### DISCUSSION THREAD

**Discussion by bddvlpr** [vercel/community#6172] [Help] [answered]
### Summary

While writing my SvelteKit app (using `vercel dev`, I noticed it would start denying form posts as the origin differs from the server's set origin.
```
Cross-site POST form submissions are forbidden
```

After a bit of digging around, I noticed that when running the app using `vercel dev --debug` the command that actually starts SvelteKit is as follows; `Spawning dev command: vite dev --port 57886`. This is what (I think) breaks the form posts as SvelteKit might be looking for an origin header that follows the default URL (http://localhost:5173). This ended up also breaking any callbacks when using AuthJS as it would randomly redirect me to `[::1]:<temp dev port>` after logging in on the remote server.

Is there a way to make the Vercel CLI not use a proxy or disable this behaviour?

### Example

https://github.com/bddvlpr/vercel-origin-fail

### Steps to Reproduce

1. Clone the example repository above
2. Run `pnpm i` and `vercel dev` to start a dev server
3. Go to http://localhost:3000/ and submit a random value

**Comment by TooTallNate** [comment, marked as answer]
That is indeed how the `vercel dev` command works - it spawns your project's Dev Command as a child process and layers Vercel-specific functionality like the `api` directory and routes from `vercel.json` on top of that.

If you are not relying on said Vercel-specific functionality, you probably don't even need to run `vercel dev`. Just run the `vite dev` server directly.

However, if you are relying on Vercel-specific functionality, then the solution is going to be to make your dev server "proxy aware". Request headers such as `x-forwarded-host` and `x-forwarded-port` and sent for every request, which the dev server should respect when doing redirects and such.

**Reply by bddvlpr** [reply]
I'm looking to migrate projects to Vercel Postgres so using the vite dev server itself wouldn't be a choice. I'll check if I can maybe write a middleware to use the request headers instead. Thanks for the quick reply.

**Reply by TooTallNate** [reply]
I don't think you'd need `vercel dev` to use Vercel Postgres. You need to pull the environment variables for your project and then load them into the `vite dev` server. Take a look at the [`vercel env pull` command](https://vercel.com/docs/cli/env).

**Reply by bddvlpr** [reply]
I tried that before but it seems like the env variable doesn't get loaded. I checked and .env (and) .env.local both have the POSTGRES_URL variable in them.
<img width="1051" alt="image" src="https://github.com/vercel/community/assets/17461028/220aaf84-bb4c-4246-93cd-57d56bf97183">

**Reply by bddvlpr** [reply]
Just found out about https://github.com/vercel/storage/tree/main/packages/postgres#a-note-for-vite-users, sorry.
